agnostic/cli.py

```diff
@@ -89,6 +89,14 @@ def __init__(self):
     type=click.Path(exists=True),
     help='Path to migrations directory. (default: ./migrations)'
 )
+@click.option(
+    '-k', '--private-key',
+    envvar='SNOWFLAKE_PRIVATE_KEY',
+    metavar='<private_key>',
+    required=False,
```

Comment:
At first I did something like this:

https://stackoverflow.com/questions/44247099/click-command-line-interfaces-make-options-required-if-other-optional-option-is/44349292

But then it seemed like it'll get awfully confusing because they're only mutually exclusive for snowflake and not the other options, so instead i made it optional but will default to a private key if you have it

Feel free to let me know if you have a different preference